recipes/boost/all/test_package/conanfile.py

```diff
@@ -1,94 +1,101 @@
+from conan import ConanFile
+from conan.errors import ConanException
+from conan.tools.build import can_run
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.scm import Version
+from conans import tools as tools_legacy
 import os
 
-from conan.tools.build import cross_building
-from conans import ConanFile, CMake, tools
-from conans.errors import ConanException
-
 
 class TestPackageConan(ConanFile):
-    settings = "os", "compiler", "arch", "build_type"
-    generators = "cmake", "cmake_find_package"
-
-    def build_requirements(self):
-        if self.settings.os == "Macos" and self.settings.arch == "armv8":
-            # Attempting to use @rpath without CMAKE_SHARED_LIBRARY_RUNTIME_C_FLAG being
-            # set. This could be because you are using a Mac OS X version less than 10.5
-            # or because CMake's platform configuration is corrupt.
-            self.build_requires("cmake/3.20.1")
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "CMakeDeps", "VirtualRunEnv"
+    test_type = "explicit"
 
     def _boost_option(self, name, default):
         try:
             return getattr(self.options["boost"], name, default)
         except (AttributeError, ConanException):
             return default
 
+    def layout(self):
+        cmake_layout(self)
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["HEADER_ONLY"] = self.dependencies["boost"].options.header_only
+        if not self.dependencies["boost"].options.header_only:
+            tc.variables["Boost_USE_STATIC_LIBS"] = not self.dependencies["boost"].options.shared
+        tc.variables["WITH_PYTHON"] = not self.dependencies["boost"].options.without_python
+        if not self.dependencies["boost"].options.without_python:
+            pyversion = Version(self.dependencies["boost"].options.python_version)
+            tc.variables["Python_ADDITIONAL_VERSIONS"] = f"{pyversion.major}.{pyversion.minor}"
+            tc.variables["PYTHON_COMPONENT_SUFFIX"] = f"{pyversion.major}.{pyversion.minor}"
+        tc.variables["WITH_RANDOM"] = not self.dependencies["boost"].options.without_random
+        tc.variables["WITH_REGEX"] = not self.dependencies["boost"].options.without_regex
+        tc.variables["WITH_TEST"] = not self.dependencies["boost"].options.without_test
+        tc.variables["WITH_COROUTINE"] = not self.dependencies["boost"].options.without_coroutine
```

Comment:
I tried testing this one and it does not like me either

```log
boost/1.80.0 (test package): Calling build()
boost/1.80.0 (test package): CMake command: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/Users/christopherm/conan-center-index/recipes/boost/all/test_package/build/generators/conan_toolchain.cmake" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/Users/christopherm/conan-center-index/recipes/boost/all/test_package/."
-- Using Conan toolchain: /Users/christopherm/conan-center-index/recipes/boost/all/test_package/build/generators/conan_toolchain.cmake
-- Conan: Component target declared 'Boost::diagnostic_definitions'
-- Conan: Component target declared 'Boost::disable_autolinking'
-- Conan: Component target declared 'Boost::dynamic_linking'
-- Conan: Component target declared 'Boost::headers'
-- Conan: Component target declared 'Boost::boost'
-- Conan: Component target declared 'boost::_libboost'
-- Conan: Component target declared 'Boost::atomic'
-- Conan: Component target declared 'Boost::container'
-- Conan: Component target declared 'Boost::context'
-- Conan: Component target declared 'Boost::date_time'
-- Conan: Component target declared 'Boost::exception'
-- Conan: Component target declared 'Boost::program_options'
-- Conan: Component target declared 'Boost::regex'
-- Conan: Component target declared 'Boost::serialization'
-- Conan: Component target declared 'Boost::stacktrace'
-- Conan: Component target declared 'Boost::stacktrace_addr2line'
-- Conan: Component target declared 'Boost::stacktrace_backtrace'
-- Conan: Component target declared 'Boost::stacktrace_basic'
-- Conan: Component target declared 'Boost::stacktrace_noop'
-- Conan: Component target declared 'Boost::system'
-- Conan: Component target declared 'Boost::test'
-- Conan: Component target declared 'Boost::test_exec_monitor'
-- Conan: Component target declared 'Boost::wserialization'
-- Conan: Component target declared 'Boost::chrono'
-- Conan: Component target declared 'Boost::filesystem'
-- Conan: Component target declared 'Boost::prg_exec_monitor'
-- Conan: Component target declared 'Boost::random'
-- Conan: Component target declared 'Boost::thread'
-- Conan: Component target declared 'Boost::timer'
-- Conan: Component target declared 'Boost::type_erasure'
-- Conan: Component target declared 'Boost::unit_test_framework'
-- Conan: Component target declared 'Boost::contract'
-- Conan: Component target declared 'Boost::iostreams'
-- Conan: Component target declared 'Boost::locale'
-- Conan: Component target declared 'Boost::log'
-- Conan: Component target declared 'Boost::log_setup'
-- Conan: Target declared 'boost::boost'
-- Conan: Target declared 'BZip2::BZip2'
-- Conan: Including build module from '/Users/christopherm/.conan/data/bzip2/1.0.8/_/_/package/f2848f95be36a5d770fc8e833993b621f703b507/lib/cmake/conan-official-bzip2-variables.cmake'
-- Conan: Target declared 'ZLIB::ZLIB'
-- Conan: Target declared 'Iconv::Iconv'
-- Conan: Target declared 'libbacktrace::libbacktrace'
-- Configuring done
CMake Error at CMakeLists.txt:30 (target_link_libraries):
  Target "coroutine_exe" links to:

    Boost::coroutine

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
ERROR: boost/1.80.0 (test package): Error in build() method, line 55
	cmake.configure()
	ConanException: Error 1 while executing cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/Users/christopherm/conan-center-index/recipes/boost/all/test_package/build/generators/conan_toolchain.cmake" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/Users/christopherm/conan-center-index/recipes/boost/all/test_package/."
 ✘ christopherm@christopherm-mac  ~/conan-center-index/recipes/boost   boost-conan-v2  conan create all boost/1.80.0@  -o boost:without_coroutine=True
```

Reply:
I can't reproduce this error on my mac, I guess you have forgotten to clear build folder of test package.